File: dongyimai-sellergoods-interface/src/main/java/com/senjie/sellergoods/service/BrandService.java
package com.senjie.sellergoods.service;

import com.senjie.entity.PageResult;
import com.senjie.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * @Author SenJie
 * @Data 2021/3/30 16:04
 */
public interface BrandService {

    /**
     * View all brands
     **/
    public List<TbBrand> findAllBrands();

    /**
     * @param pagNum  current page
     * @param pagSize page size
     * @return
     */
    public PageResult findPage(int pagNum, int pagSize);

    /**
     * The brand add
     *
     * @param tbBrand
     */

    public void add(TbBrand tbBrand);

    /**
     *   Query by id
     * @param id
     * @return
     */


    public TbBrand findOne(Long id);

    /**
     * Update
     * @param brand
     */


    public  void  update(TbBrand brand);

    /**
     * Delete by ids
     * @param ids
     */

    public  void delete(Long [] ids);

    /**
     *  fuzzy Query
     * @param brand
     * @param pageNum
     * @param pagSize
     * @return
     */

    public  PageResult  findPage(TbBrand brand,int pageNum, int pagSize);


    public List<Map>  findBrandList();

}

File: dongyimai-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {

    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };
    /**
     *
     * @param page
     * @param rows
     * @returns {*}
     */
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };
    /**
     *
     * @param entity
     * @returns {*}
     */
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }
    /**
     *
     * @returns {*}
     */
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }

    /**
     *
     * @param id
     * @returns {*}
     */
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }
    /**
     *
     * @returns {*}
     */
    this.delete = function (ids) {
        return $http.get('../brand/delete.do?id=' + ids);
    }
    /**
     *
     * @param page
     * @param rows
     * @param serchEntity
     * @returns {*}
     */
    this.search = function (page, rows, serchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&rows=' + rows, serchEntity);
    }

    this.finds = function () {
        return $http.get('../brand/findBrandList.do');
    }

})